Source/Core/Core/Boot/Boot.cpp

```diff
@@ -300,19 +308,46 @@ bool CBoot::LoadMapFromFilename()
   return false;
 }
 
+static void RAMOverrideForIPLMemoryValues(std::string& data, const u32 ipl_hash)
+{
+  if (!Config::Get(Config::MAIN_RAM_OVERRIDE_ENABLE))
+    return;
+
+  // Pre-calculate the new values.
+  const s16 mem1_s_size = Common::swap16(Config::Get(Config::MAIN_MEM1_SIZE) >> 16);
+  const u32 nop = Common::swap32(0x60000000);
+
+  // [lis	r0, 0x0180] <-- This instruction is what sets the Simulated MEM1 Size in retail IPLs.
+  // The following code replaces the SIMM field and nothing else.
+  // [ori	r9, r9, 0x0003] <-- This instruction putting the value "3" into r9 causes a stw later
+  // down the line to revert the instruction we're trying to change.  I don't understand why, but
+  // I do know that NOPing this away doesn't break anything!
```

Comment:
It appears that `r9` is then written to [EXI0CR](http://hitmen.c02.at/files/yagcd/yagcd/chap5.html#sec5) (`0xCC00680C`), where 3 is start (1) DMA (2) read.  Setting this to 0 disables it (and since the code then polls EXI0CR for bit 0 to be cleared, it immediately finishes).

But perhaps it would be better to modify the data it's reading from EXI instead of patching it out?  I'm not sure where it'd be getting it from even (maybe it's something in `EXI_DeviceIPL`, but I don't see anything that explicitly implements it).

Reply:
Finally some closure on what the heck was going on.  I was so confused watching as a stw instruction writing to some bizarre address was causing instructions to revert of all things.

Reply:
... in fact, the whole `Memory::CopyToEmu(0x01300000, data.data() + 0x820, 0x1AFE00);` line is unneeded, since BS1 does that itself ([see YAGCD](http://hitmen.c02.at/files/yagcd/yagcd/chap18.html#sec18.2), though when they say "clearing bit 17 of 0xcc006800" it looks like it should be bit 13 (`ROMDIS`)).

But I don't think you need to address that in this PR; just change the comment here to something like

> // [ori	r9, r9, 0x0003] <-- The value 3 in r9 later starts an EXI DMA read to copy BS2, which clobbers the patched version.  Leaving it at 0 makes it think the EXI transfer finished immediately, skipping it.